### DISCUSSION TITLE
Does async validation (eg async resolver) clear error state?

### DISCUSSION THREAD

**Discussion by riotrah** [react-hook-form/react-hook-form#10028] [Q&A]
Hi, thanks for this lib, it's wonderful. I tried to search for similar issues but couldn't find my exact issue.

Because RHF supports async validation (eg for `resolver`), I'm wondering: does RHF automatically will clear the error state when the async validation _starts_? If not, what's the best time or API to implement that for all registered fields?

**Comment by bluebill1049** [comment]
I think we do that for the submission, but for individual field updates, it will have to wait for the promise to resolve before we update the errors state.
